nsist/pypi.py

```diff
@@ -250,12 +252,83 @@ def extract_wheel(whl_file, target_dir, exclude=None):
     shutil.rmtree(str(td))
 
 
-def fetch_pypi_wheels(requirements, target_dir, py_version, bitness,
-                      extra_sources=None, exclude=None):
-    for req in requirements:
+def fetch_pypi_wheels(wheels_requirements, wheels_paths, target_dir, py_version,
+                      bitness, extra_sources=None, exclude=None):
+    """
+    Gather wheels included explicitly by wheels_pypi parameter 
+    or matching glob paths given in local_wheels parameter.
+    """
+    wheel_info_array = []
+    # We try to get the wheels from wheels_pypi requirements parameter
+    for req in wheels_requirements:
         wl = WheelLocator(req, py_version, bitness, extra_sources)
-        whl_file = wl.fetch()
+        whl_file = wl.fetch() 
+        validate_wheel(whl_file, wheel_info_array, py_version)
         extract_wheel(whl_file, target_dir, exclude=exclude)
+    # Then from the local_wheels paths parameter
+    for glob_path in wheels_paths:
+        for path in glob.glob(glob_path):
```

Comment:
There should probably be at least a warning if a glob pattern matches no files at all - there's a good chance this means something has gone wrong with an earlier build step.

And unfortunately, warnings aren't very visible, because once we launch `makensis`, it rapidly produces a lot of output, and our own logging is pushed off the screen. So it might be best to make it an error and see if there's an actual use case.

(Working out if we should hide makensis output is a separate topic - I don't want to change that in this PR)